File: go/scrabble-score/scrabble_score.go
// Package scrabble score calculates a scrabble score
package scrabble

// Score calculates a scrabble score
func Score(word string) (score int) {
	for _, char := range word {
		switch char {
		case 'A', 'E', 'I', 'O', 'U', 'L', 'N', 'R', 'S', 'T', 'a', 'e', 'i', 'o', 'u', 'l', 'n', 'r', 's', 't':
			score++
		case 'D', 'G', 'd', 'g':
			score += 2
		case 'B', 'C', 'M', 'P', 'b', 'c', 'm', 'p':
			score += 3
		case 'F', 'H', 'V', 'W', 'Y', 'f', 'h', 'v', 'w', 'y':
			score += 4
		case 'K', 'k':
			score += 5
		case 'J', 'X', 'j', 'x':
			score += 8
		case 'Q', 'Z', 'q', 'z':
			score += 10
		}
	}
	return
}

File: python/scrabble-score/scrabble_score.py
score_map = {
    letter: points
    for letters, points in [
        ("A, E, I, O, U, L, N, R, S, T", 1),
        ("D, G", 2),
        ("B, C, M, P", 3),
        ("F, H, V, W, Y", 4),
        ("K", 5),
        ("J, X", 8),
        ("Q, Z", 10),
    ]
    for letter in letters.split(", ")
}


def score(word):
    return sum(score_map.get(letter, 0) for letter in word.upper())
